app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -435,7 +435,7 @@ export default class ComposerService extends Service {
 
       return options.concat(
         customPopupMenuOptions
-          .map((option) => this._setupPopupMenuOption(option))
+          .map((option) => this._setupPopupMenuOption({ ...option }))
```

Comment:
So that this logic doesn't cause the original function to be overwritten

https://github.com/discourse/discourse/blob/29752dda2ee75ed14928b6e2fd74be864bfd65d4/app/assets/javascripts/discourse/app/services/composer.js#L365-L366